beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/historical/HistoricalBatchFetcher.java

```diff
@@ -239,10 +241,28 @@ private SafeFuture<Boolean> requestBlocksAndBlobSidecarsByRange() {
         .thenApply(__ -> shouldRetryBlocksAndBlobSidecarsByRangeRequest());
   }
 
+  // We explicitly mark post-Deneb slots in availability range as some could contain no BlobSidecars
+  private void fillBlobSidecarBySlotToImport(final UInt64 startSlot, final UInt64 endSlot) {
```

Comment:
Hmm that works yeah, but I wonder if logic can be simplified to avoid all this loops pre-Deneb. Shall we discuss on Monday?

Reply:
it's called only when we are inside if `    if (blobsSidecarManager.isAvailabilityRequiredAtSlot(endSlot)) {`

Reply:
kk just rename to `fillBlobSidecarsBySloToImport`